pages/_app.tsx

```diff
@@ -1,11 +1,42 @@
+import useLocalStorageState from 'use-local-storage-state';
 import GlobalStyle from '../styles';
 import type { AppProps } from 'next/app';
+import recipes from '@/lib/recipes.json';
 
 export default function App({ Component, pageProps }: AppProps) {
+  const [favoriteRecipesList, setFavoriteRecipesList] = useLocalStorageState(
+    'favoriteRecipesList',
+    { defaultValue: [] }
+  );
+
+  const handleToggleFavorite = (id) => {
```

Comment:
The type of `id` currently gets inferred, but you could be more explicit here:

```suggestion
  const handleToggleFavorite = (id: string) => {
```